charts/datahub/subcharts/datahub-frontend/templates/deployment.yaml

```diff
@@ -15,10 +15,13 @@ spec:
       {{- include "datahub-frontend.selectorLabels" . | nindent 6 }}
   template:
     metadata:
-    {{- with .Values.podAnnotations }}
       annotations:
```

Comment:
Have you dry-run these changes?
By moving this `with` statement you are rendering an `annotations` block without content if they are no user-provided annotations and if `defaultUserCredentials` is not used.

Reply:
We would like to keep the generated files as clean as possible. Hence the comment :)

Reply:
Hey @pedro93, thanks for the comment! 
Yes, I was aware of this. My reasoning was that an empty `annotations:` block shouldn't do any harm and the logic for checking whether or not to print it would become more and more complex the more annotation sources are added. But I totally understand that you want to keep it tidy, so I'll update it accordingly :)

Reply:
@pedro93 I've updated the branch and surrounded everything with an `{{- if or .Values.podAnnotations .Values.defaultUserCredentials }}`.

Alternatively, I could propose this:
```yaml
    metadata:
      {{- $annotations := merge (dict) .Values.podAnnotations }}
      {{- if .Values.defaultUserCredentials }}
      {{- $annotations := merge $annotations (dict "checksum/secret" (include (print $.Template.BasePath "/user-secrets.yaml") . | sha256sum)) }}
      {{- end}}
      {{- with $annotations }}
      annotations:
        {{- toYaml . | nindent 8 }}
      {{- end }}
```

But neither of these solutions seem much better to me than the previous state. Plus: there are already numerous occurrences of potentially empty keys (e.g. [volumes](https://github.com/acryldata/datahub-helm/blob/6311fce06c11ce21c5c3edabf68b71fff88c027c/charts/datahub/subcharts/datahub-frontend/templates/deployment.yaml#L38), [initContainers](https://github.com/acryldata/datahub-helm/blob/6311fce06c11ce21c5c3edabf68b71fff88c027c/charts/datahub/subcharts/datahub-frontend/templates/deployment.yaml#L54), [volumeMounts](https://github.com/acryldata/datahub-helm/blob/6311fce06c11ce21c5c3edabf68b71fff88c027c/charts/datahub/subcharts/datahub-frontend/templates/deployment.yaml#L237) to name a few). 

Anyways, I'm happy with implementing whatever you prefer, as long as the functionality is there 😉

Reply:
I see what you mean, not happy with either version. Can you revert to the original proposed version? I’ll approve that one

Reply:
Alright, done :)